File: api/src/main/java/org/example/api/service/WorkoutDurationService.java
package org.example.api.service;

import org.example.api.dto.WorkoutDurationRecordRequestDTO;
import org.example.api.dto.WorkoutDurationSummaryDTO;

import java.time.LocalDate;

public interface WorkoutDurationService {
    void recordDuration(WorkoutDurationRecordRequestDTO requestDTO);

    WorkoutDurationSummaryDTO getSummary(LocalDate startDate, LocalDate endDate);
}

File: healthy/app/src/main/java/com/example/myapplication/data/remote/WorkoutDurationService.kt
package com.example.myapplication.data.remote

import com.example.myapplication.data.model.WorkoutDurationRecordRequest
import com.example.myapplication.data.model.WorkoutDurationSummary
import retrofit2.http.Body
import retrofit2.http.GET
import retrofit2.http.POST
import retrofit2.http.Query

interface WorkoutDurationService {
    @POST("workoutDuration/record")
    suspend fun recordDuration(@Body request: WorkoutDurationRecordRequest)

    @GET("workoutDuration/summary")
    suspend fun getSummary(
        @Query("startDate") startDate: String,
        @Query("endDate") endDate: String
    ): WorkoutDurationSummary
}
